### PR TITLE
Translatable Main Panel

### PR DESCRIPTION
**Pull request by AJenbo**
image
image

### CODE REVIEW

**Comment by julealgon** [comment]
Is that original screenshot old? The XP bar design is different.

Would be nice if you could capture a more recent version for comparison purposes.

**Comment by AJenbo** [comment]
> Is that original screenshot old? The XP bar design is different.
> 
> Would be nice if you could capture a more recent version for comparison purposes.

The screenshot is of 1.2.0, the life and mana level is different, a different spell is prepared, one has the multiplayer buttons, the monsters in the background are also in a different state. Here is one without all the distractions:
image

**Comment by AJenbo** [comment]
Also in case no one noticed, there is a bug in the original art:
image

**Review by obligaron — COMMENTED** [review]
lgtm, but I'm not familiar with the rendering functions.
I had to search that `DrawArt` does nothing if `Art` is not initialized.

Side note: with this changes we could fix that in **debug** singleplayer we draw the player friendly toggle button.

File: Source/control.cpp [hunk 1]
```diff
@@ -898,6 +901,97 @@ void control_update_life_mana()
...
+	UpdatePalette(&PanelButtonDown);
+	if (SDLC_SetSurfaceColors(pBtmBuff->surface, PalSurface->format->palette) <= -1)
+		ErrSdl();
+
+	RenderMainButton(0, _("Char"), 0);
+	RenderMainButton(1, _("Quests"), 1);
+	RenderMainButton(2, _("Map"), 1);
+	RenderMainButton(3, _("Menu"), 0);
+	RenderMainButton(4, _("Inv"), 1);
+	RenderMainButton(5, _("Spells"), 0);
```
[the remark this answers is not in the document]
Comment (AJenbo) [reply Source/control.cpp:966]: Yes, and yes. That's how I was able to make a Portuguese screen shot :)

To not share translations we would need to fix https://github.com/diasurgical/devilutionX/issues/2790 but i don't think it's relevant for this case.

File: Source/control.cpp [hunk 2]
```diff
@@ -930,6 +1024,8 @@ void InitControlPan()
 		for (bool &talkButtonDown : TalkButtonsDown)
 			talkButtonDown = false;
 	}
+	if (strcmp(sgOptions.Language.szCode, "en") != 0)
+		RenderButtonsPanel();
```
[the remark this answers is not in the document]
Comment (AJenbo) [reply Source/control.cpp:1028]: you could say the same for a translation that translates everything except the button texts. The lang detection will fallback to en if we do not have a translation for the given system default, so in most cases this should be ok.

**Comment by AJenbo** [comment]
Art department (@NikoVP)  rejected the PR, stand by for update:

image

**Comment by julealgon** [comment]
> > Is that original screenshot old? The XP bar design is different.
> > Would be nice if you could capture a more recent version for comparison purposes.
> 
> The screenshot is of 1.2.0, the life and mana level is different, a different spell is prepared, one has the multiplayer buttons, the monsters in the background are also in a different state. Here is one without all the distractions

Thanks. I actually noticed this because of the difference in the experience bar. We don't have the "XP" text next to it anymore?

I explicitly noticed that since it is also text, and would be sensitive to translations,

**Comment by AJenbo** [comment]
> > > Is that original screenshot old? The XP bar design is different.
> > > Would be nice if you could capture a more recent version for comparison purposes.
> > 
> > 
> > The screenshot is of 1.2.0, the life and mana level is different, a different spell is prepared, one has the multiplayer buttons, the monsters in the background are also in a different state. Here is one without all the distractions
> 
> Thanks. I actually noticed this because of the difference in the experience bar. We don't have the "XP" text next to it anymore?
> 
> I explicitly noticed that since it is also text, and would be sensitive to translations,

The xp bar was revamped shortly after the previous release

**Comment by AJenbo** [comment]
PR is now updated with the latest version of the panel and ready for review.

Alternate to `ColorUiGoldDark`, `ColorUiSilverDark`, `ColorWhite`, and `ColorWhitegold`, i could make it a special case for size12 in the LoadFont function.

File: Source/control.cpp [hunk 3]
```diff
@@ -898,6 +901,97 @@ void control_update_life_mana()
 	Players[MyPlayerId].UpdateHitPointPercentage();
 }
 
+namespace {
```
[the remark this answers is not in the document]
Comment (AJenbo) [reply Source/control.cpp]: i moved it to it's own file to mirror charpanel.cpp

**Review by StephenCWills — APPROVED** [review]
I just found a little typo.

**Comment by StephenCWills** [comment]
Oops, I was like 20 seconds too late.

**Comment by AJenbo** [comment]
It was on automerge :)

**Pull request merged**